sympy/core/function.py

```diff
@@ -1253,24 +1253,7 @@ def __new__(cls, expr, *variables, **assumptions):
                     expr = expr.xreplace({v: new_v})
                     old_v = v
                     v = new_v
-                obj = expr
-                if count.is_Integer:
-                    for i in range(count):
-                        obj2 = deriv_fun(obj, v)
-                        if obj == obj2:
-                            break
-                        obj = obj2
-                        nderivs += 1
-                elif obj.is_Derivative:
-                    dict_var_count = dict(obj.variable_count)
-                    if v in dict_var_count:
-                        dict_var_count[v] += count
-                    else:
-                        dict_var_count[v] = count
-                    obj = Derivative(obj.expr, *dict_var_count.items())
-                    nderivs += count
-                else:
-                    obj = None
+                obj = expr._eval_derivative_n_times(v, count)
```

Comment:
I think fallback to `_helper_apply_n_times` should be done here.

Reply:
I like it better as a fallback in the default `_eval_derivative_n_times`, at least with this API where it's a separate function from `_eval_derivative`.